sparse/_common.py

```diff
@@ -438,3 +438,33 @@ def outer(a, b, out=None):
            [0, 3, 6, 9]])
     """
     return np.multiply.outer(a, b, out=out)
+
+
+def _is_dask_collection(x):
+    """
+    Returns ``True`` if ``x`` is a dask collection
+
+    Vendored from dask.base
+    """
+    try:
+        return x.__dask_graph__() is not None
+    except (AttributeError, TypeError):
+        return False
+
+
+def asarray(a, dtype=None, order=None):
+    """Returns a dense numpy array from an arbitrary source array.
+
+    Args:
+        a: Arbitrary object that can be converted to :class:`numpy.ndarray`.
+        order ({'C', 'F', 'A'}): The desired memory layout of the output
+            array. When ``order`` is 'A', it uses 'F' if ``a`` is
+            fortran-contiguous and 'C' otherwise.
+    Returns:
+        numpy.ndarray: Converted array on the host memory.
+    """
+    from ._sparse_array import SparseArray
+
+    if isinstance(a, SparseArray):
+        a = a.todense()
+    return np.array(a, dtype=dtype, copy=False, order=order)
```

Comment:
Should be `np,asarray` because `np.array` always copies, which is wasteful.

Reply:
I believe that `copy=False` prevents the copy, as verified by this part of the test:

```py
    a = np.array([1, 2, 3])
    # Array passes through with no copying.
    assert sparse.asarray(a) is a
```